lib/src/connector/connector.cc

```diff
@@ -723,13 +745,18 @@ void Connector::startMonitorTask(const uint32_t max_connect_attempts,
             // Associate Session failure - stop
             LOG_WARNING("The connection monitor task will stop after an "
                         "Session Association failure: {1}", e.what());
-            is_monitoring_ = false;
-            throw;
+            monitor_exception_ = std::current_exception();
+            break;
```

Comment:
In case multiple brokers are configured, we want to switch broker after `connection_associate_error` or `connection_fatal_error` here, instead of put the user code in charge of retrying with new `connect` and `monitorConnection calls`; the reason is, since `connect` switches brokers if needed, `monitorConnection` should as well.
 
Also, in the multiple brokers case, we may want to revisit the usage of `max_connect_attempts`.

I'll make a ticket for this.